lib/mint/http2.ex

```diff
@@ -478,15 +478,9 @@ defmodule Mint.HTTP2 do
     headers =
       headers
       |> downcase_header_names()
+      |> add_pseudo_headers(conn, method, path)
```

Comment:
We don't do any validation on the method being all upcase IIRC, so let's do this here maybe?

```suggestion
      |> add_pseudo_headers(conn, String.upcase(method), path)
```